developers/ide/eclipse.md

```diff
@@ -5,65 +5,153 @@ title: Eclipse IDE
+   As a result when you restart Eclipse the other day. It may take some time as it updates all SNAPSHOT versions of openHAB.
+
+1. If you want to develop a new binding. Read about the [skeleton script](../#develop-a-new-binding) to generates the basis for your binding, creating all required files.
+
+1. If you need additional libraries see the [Build System](../buildsystem.html) documentation.
+For other libraries supported out-of-the-box check the [Default Libraries](../guidelines.html#default-libraries) on the guidelines page.
+
+## Debugging an Addon
 
-    When visible in the Package Explorer, select `OH2 Add-ons` and from right-click menu select "Close Projects": this significantly speeds up the setup.
-    Re-open only the binding project(s) you would like to work on.
+To test your binding you can build the addon on the command line with maven and drop the jar file in an addons folder of an existing installation.
+But to easily debug an addon the `openHAB Development` setup installs and imports a demo project that contains a complete openHAB environment to run and debug an addon.
 
-    **It may take 30-45mins for all initial tasks to finish, depending on internet connection and your computer speed.
-    So... go and grab a good cup of coffee and be patient until is finished**
+Under `Infrastructure` you'll find the project `org.openhab.demo.app`.
+This project contains the full configuration to start a debug sessions.
+The following files are of interest for the debug environment:
 
-   ![startup tasks finished](./images/ide_setup_eclipse_tasks_finished.png)
+```
+|- org.openhab.demo.app
+|--- runtime
+|------- conf             Here you configure the manual text files
+|------- userdata         Here is the openHAB user data
+|------- logback.xml      XML file for logging options
+|--- app.bndrun           The file to start a debug session
+|--- pom.xml              The pom file with all dependencies for the demo project
+```
 
-1. After all tasks are finished, look for `app.bndrun` file in the Eclipse Package Explorer in `Infrastructure > launch > app` project folder.
-   Double click to open `app.bndrun` file (takes a few seconds)
+1. Import the addon in Eclipse.
+Either it's an existing addon or a new binding created with the skeleton script.
+Import the addon project via `File > Import... > General > Existing Projects into Workspace`.
+Importing an addon is necessary if you want to modify or debug the addon.
+It's also possible to run existing addons part of the SNAPSHOT release in the demo project without having it imported in Eclipse.
+Simply follow the next step to add the addon.
 
-1. Under Browse Repos search for the binding you want to run/debug (`astro` in our case) and add it to the "Run Requirements" list using drag&drop from the "Browse Repos" list:
-    ![Bndtools](images/ide_setup_eclipse_bndtools.png)   
+1. To let the demo project know about the addon, the addon must be added to the demo project `pom.xml`.
```

Comment:
What I faced was that I also had to manually add the dependency of the project (which makes sense because it was marked `provided`) but it might still be a good idea that does do not work because of its does not use the feature files. 

BTW: Was that provided already dropped as the prefered solution or not?

Reply:
@Hilbrand did you address this?

Reply:
@martinvw yes I think provided is dropped. A number of bindings (if not all) have been migrated to `compile`. I probably will check if there needs some additional information when working with additional libraries. But for now I linked to the documentation page containing information about how to include libraries.

Reply:
:-(

https://github.com/openhab/openhab2-addons/blob/bf638391e58ceaecad7a8623e43a1251881b6a7e/bundles/org.openhab.binding.heos/pom.xml#L21

Reply:
So not completely solved, but it should be a corner case now.